jupyter_releaser/lib.py

```diff
@@ -162,29 +159,58 @@ def make_changelog_pr(auth, branch, repo, title, commit_message, body, dry_run=F
     head = pr_branch
     maintainer_can_modify = True
 
-    if not dry_run:
-        util.run(f"git push origin {pr_branch}", echo=True)
+    util.run(f"git push origin {pr_branch}")
 
-    #  title, head, base, body, maintainer_can_modify, draft, issue
+    # title, head, base, body, maintainer_can_modify, draft, issue
+    util.log('Creating a PR"')
     pull = gh.pulls.create(title, head, base, body, maintainer_can_modify, False, None)
+    util.log(f'Created a PR: {pull.number}')
 
     # Try to add the documentation label to the PR.
     number = pull.number
+    util.log("Adding label")
     try:
-        gh.issues.add_labels(number, ["documentation"])
+        if pr_type == "forwardport":
+            gh.issues.add_labels(number, ["documentation"])
+        else:
+            gh.issues.add_labels(number, ["maintenance"])
     except Exception as e:
-        print(e)
+        util.log(e)
+    util.log("Added label")
+
+    if pr_type == "release":
+        # Merge the release PR
+        sha = util.run("git rev-parse HEAD")
+        commit_message = util.run(f"git log --format=%B -n 1 {sha}")
+        util.log(f"Merging the PR {number}")
+        gh.pulls.merge(number, title, commit_message, sha, "rebase")
```

Comment:
If I follow correctly, this will use the GH API to merge the release PR. If it fails the populate release step will fail. Is that correct?

---

What happen if pull requests have protection rule; e.g. on JupyterLab it is required that the linter job succeed. Will this line fail in that case?

Reply:
Good question, I'll test it out on my test repo.

Reply:
```
====Error Body====
{
  "message": "Required status check \"test\" is expected.",
  "documentation_url": "https://docs.github.com/articles/about-protected-branches"
}
```

Hmm, given that there is a way to do this manually on a PR, I'll have to look at the GitHub API docs.

Reply:
Okay, I think a combination of using `[ci skip]` and then a backoff until the PR can be merged might work.